cast/src/starknet_commands/deploy.rs

```diff
@@ -65,16 +70,34 @@ pub async fn deploy(
         deployment
     };
 
-    let result = execution.send().await?;
+    let result = execution.send().await;
 
-    wait_for_tx(account.provider(), result.transaction_hash).await;
-
-    let address = get_contract_address(
-        salt,
-        class_hash,
-        &raw_constructor_calldata,
-        FieldElement::from_hex_be(UDC_ADDRESS)?,
-    );
-
-    Ok((result.transaction_hash, address))
+    match result {
+        Ok(result) => match wait_for_tx(account.provider(), result.transaction_hash).await {
+            Ok(_) => Ok((
+                result.transaction_hash,
+                get_udc_deployed_address(
+                    salt,
+                    class_hash,
+                    &if unique {
+                        Unique(UdcUniqueSettings {
+                            deployer_address: account.address(),
+                            udc_contract_address: FieldElement::from_hex_be(UDC_ADDRESS)?,
+                        })
+                    } else {
+                        NotUnique
+                    },
+                    &raw_constructor_calldata,
+                ),
+            )),
+            Err(message) => Err(anyhow!(message)),
+        },
```

Comment:
I wonder if we could also abstract this snippet, to be a reusable function for our commands. what do you think?